core/src/main/java/com/linecorp/armeria/client/HttpClient.java

```diff
@@ -139,6 +141,61 @@ static HttpClient of(ClientFactory factory, URI uri, ClientOptions options) {
         return new HttpClientBuilder(uri).factory(factory).options(options).build();
     }
 
+    /**
+     * Creates a new HTTP client that connects to the specified {@link Endpoint} with {@code scheme}
+     * using the default {@link ClientFactory}.
+     *
+     * @param scheme a scheme of URI
+     * @param endpoint the server {@link Endpoint}
+     * @param options the {@link ClientOptionValue}s
+     */
+    static HttpClient of(String scheme, Endpoint endpoint, ClientOptionValue<?>... options) {
+        requireNonNull(endpoint, "endpoint");
+        return of(ClientFactory.DEFAULT, endpoint.toURI(scheme), options);
+    }
+
+    /**
+     * Creates a new HTTP client that connects to the specified {@link Endpoint} with {@code scheme}
+     * using the default {@link ClientFactory}.
+     *
+     * @param scheme a scheme of URI
+     * @param endpoint the server {@link Endpoint}
+     * @param options the {@link ClientOptions}
+     */
+    static HttpClient of(String scheme, Endpoint endpoint, ClientOptions options) {
```

Comment:
`String scheme` should be replaced with `SessionProtocol protocol`, because the serialization format is always `none` for `HttpClient`.